tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationManager.Subscription.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="RemoteConfigurationManager.Subscription.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.RemoteConfigurationManagement
+{
+    internal partial class RemoteConfigurationManager
+    {
+        private readonly List<Subscription> _subscriptions = new();
+        private List<string> _subscriptionsProductKeys = new();
```

Comment:
no because a subscription can target multiple products at once. So it wouldnt be `Dictionary<productkey,Subscription> `but `Dictionary<productkey[],Subscription>` and I dont see a list being a key for a dictionary... 
We still need the keeys apart for `RemoteConfigurationManager` to build the `RcmRequest` and tell all the keys it's interested in. And I didnt want to rebuild this list every 10 seconds we poll, but just when we add or remove a subcription, we rebuild the list of keys

Reply:
I think it would slightly simplify things if `_subscriptionsProductKeys` where a `HashSet<string>`

Reply:
> no because a subscription can target multiple products at once

Ok

> HashSet<string>

A small HashSet may be slightly less memory efficient than a list when small IIRC. But that better answers your need. I'm fine with the list even though I hate LINQ :p and even though it makes the code slightly more complicated. But I suppose this list won't change much so the less mem impact we will have the better ? 🤷  So as you wish :)

Reply:
In the end I went for the hashset in 3a2bc9d2bd36dee143c9feb6f50a721a36309a22 and removed most of the linq. 
The only thing with the hashset is on removal, I cant just remove the keys of the subscription as other subscription might be interested in the same key. If subscription A has key 1 and Subscription B also has key 1, I dont wanna to remove key 1 if A is disposed but not B, so I need to rebuild the hashset entirely from all subscriptions. 
But like you said, it should happen quite rarely to be honest. Only when ASM is activated / deactivated.